smart-contracts/wasm-chain-integration/src/utils.rs

```diff
@@ -337,6 +337,60 @@ pub fn generate_contract_schema_v2(
     }))
 }
 
+/// Tries to generate schemas for events, parameters, return values, and errors
+/// of methods for a contract with a V3 schema.
+pub fn generate_contract_schema_v3(
+    module_bytes: &[u8],
+) -> ExecResult<schema::VersionedModuleSchema> {
+    let artifact = utils::instantiate::<ArtifactNamedImport, _>(&TestHost, module_bytes)?;
+
+    let mut contract_schemas = BTreeMap::new();
+
+    for name in artifact.export.keys() {
+        if let Some(rest) = name.as_ref().strip_prefix("concordium_event_schema_") {
+            if let Some(contract_name) = rest.strip_prefix("init_") {
+                // Generate event schema
+                let function_schema_event = generate_schema_run(&artifact, name.as_ref())?;
+
+                let contract_schema = contract_schemas
+                    .entry(contract_name.to_owned())
+                    .or_insert_with(schema::ContractV3::default);
+                contract_schema.event = Some(function_schema_event);
+            }
```

Comment:
Perhaps add an explicit comment here as to why we do nothing in the else branch. Like we do below for the other part of the schema.